solarforecastarbiter/conftest.py

```diff
@@ -2115,19 +2135,34 @@ def report_text():
              ]},
              {"time_of_day_range": ["12:00", "14:00"]}
          ],
-         "metrics": ["mae", "rmse", "mbe", "s"],
+         "costs": [
+             {
+                "name": "example cost",
+                "type": "constant",
+                "parameters": {
+                    "cost": 1.0,
+                    "aggregation": "sum",
+                    "net": true
+                 }
+             }
+         ],
```

Comment:
I thought dashboard was going to present a single cost form for now but insert the json into all of the object pairs. Then this higher level costs key would not be needed. I think that would also remove the need for a few lines in `api.py`.

Reply:
Ah, see I was thinking the dashboard would initially only make one cost and then add the name for each object pair. 

I agree that `api.py` could be simplified without this, but the repetition in the json feels gross. Especially with a long series of datetimes, you're repeating a whole lot of information for no reason. And if you're constructing the json yourself to POST, make a mistake one place means you have to fix in many.

I can also see how this current structure can be easily extended in the future to allow for storing cost objects in the API and then just specifying the ID for that cost for the object pair. For example, maybe we somehow have historical CAISO EIM penalties as a reference Cost that anyone can apply in their reports.

Reply:
Ok that sounds good to me. 

One more related point... I don't see a new cost attribute on the ReportParameters object. My understanding now is that this attribute is not actually needed based on how `api.py` parses the json directly onto the cost object. But it feels a little hacky to include it in the json spec but not the python spec. I think this also makes it impossible to faithfully roundtrip a json-python-json representation of a report.

Reply:
correct that it isn't needed. Even without costs, the report can't be roundtripped without special processing from api.py because the forecasts/observations are replaced with their IDs instead of Forecast/Observation objects

Reply:
One solution that might make more sense is to add the `Cost` objects to `ReportParameters` keeping `ForecastObservation.cost` as a cost name. Then `preprocessing` can pull out the right cost for each `ProcessedForecastObservation`. This keeps `ReportParameters` and the json more aligned even if they can't be roundtripped

Reply:
One downside is that we can't process a `ForecastObservation` on it's own, but we couldn't already do that because the `filters` are at a high level.

Preference for full `Cost` in `ForecastObservation` or cost name in `ForecastObservation` and costs parameter in `ReportParameters`?

Reply:
No preference.

Reply:
went with costs in ReportParameters